hasher-matcher-actioner/hmalib/lambdas/api/api_root.py

```diff
@@ -117,21 +118,33 @@ def submit_content_request_from_s3_event_record(
 ) -> SubmitContentRequestBody:
     """
     Converts s3 event into a SubmitContentRequestBody object with a URL to the content
+
+    For partner bucket uploads the content IDs are unique and human understandable but
+    not reversable
+      * uniqueness is provided by uuid4 which has a collision rate of 2^-36
+      * human understandbility is provided by including the (slightly modified) key
+        in the content id
+      * modifications to the key mean that the original content bucket and key are
+        not derivable from the content ID alone
+
+    The original contnet (bucket and key) is stored in the reference url which is passed
+    to the webhook via additional_fields
     """
-    # For partner buckets we use the full bucket name and key as the content ID to avoid collisions with
-    # existing objects
-    bucket = record["bucket"]["name"]
-    key = record["object"]["key"]
-    content_id = bucket + "/" + key
+    bucket: str = record["bucket"]["name"]
+    key: str = record["object"]["key"]
+
+    readable_key = key.replace("/", ".").replace("?", ".").replace("&", ".")
```

Comment:
nit: I feel like the full path is a bit overkill here, maybe just the name after all the prefixes?